dashi/rnaseqc/over_time.py

```diff
@@ -51,72 +75,231 @@
     'Proportion UTR Bases',
 )
 
+# The colours that are used in the graphs
 COLOURS = [
     '#1f77b4', '#aec7e8', '#ff7f0e', '#ffbb78', '#2ca02c',
     '#98df8a', '#d62728', '#ff9896', '#9467bd', '#c5b0d5',
     '#8c564b', '#c49c94', '#e377c2', '#f7b6d2', '#7f7f7f',
     '#c7c7c7', '#bcbd22', '#dbdb8d', '#17becf', '#9edae5'
 ]
 
+# The shapes of the points to use in the graph
+# More can be added if the future needs it
+# See https://plot.ly/python/reference/#scatter-marker
+SHAPES = [
+    "circle", "square", "diamond", "cross", "x", "triangle-up", "triangle-down",
+    "triangle-left", "triangle-right", "pentagon", "star",
+]
+
+# Which columns will the data table always have
+DEFAULT_TABLE_COLUMN = [
+    {'name': 'Library', 'id': 'Sample Name'},
+    {'name': 'Project', 'id': 'Study Title'},
+    {'name': 'Run', 'id': 'Sequencer Run Name'},
+    {'name': 'Lane', 'id': 'Lane Number'},
+    {'name': 'Kit', 'id': 'geo_prep_kit'},
+    {'name': 'Library Source Type', 'id': 'geo_library_source_template_type'},
+    {'name': 'Tissue Origin', 'id': 'geo_tissue_origin'},
+    {'name': 'Tissue Type', 'id': 'geo_tissue_type'},
+    {'name': 'Tissue State', 'id': 'geo_tissue_preparation'},
+]
+
+# Columns on which shape and colour can be set
+SHAPE_COLOUR_COLUMN = [
+    {'name': 'Project', 'id': 'Study Title'},
+    {'name': 'Kit', 'id': 'geo_prep_kit'},
+    {'name': 'Library Source Type', 'id': 'geo_library_source_template_type'},
+    {'name': 'Tissue Origin', 'id': 'geo_tissue_origin'},
+    {'name': 'Tissue Type', 'id': 'geo_tissue_type'},
+    {'name': 'Tissue State', 'id': 'geo_tissue_preparation'},
+]
+
+# A convenience container that links how a metric should be graphed
+# The name is the column name
+# The properties is a dict
+# The dict key is a value found in the column
+# The dict value is the property (color, shape, etc) assigned to the dict keY
+PlotProperty = collections.namedtuple('PlotProperty', ['name', 'properties'])
+
+
+def assign_consistent_property(values: list, prop: list) -> dict:
+    """
+    Values in the DataFrame (project, tissue, kit) need to be assigned
+    consistent properties (color, shape). This does is fairly simply, by looping
+    in order through the values and assigning them the next available property.
+    Property lists cannot run out, as it loops back to the beginning.
+
+    Args:
+        values: The values of assign properties to. Need to be unique
+        prop: The properties to assign to
 
-def create_plot_dict(df, variable, colours=COLOURS, show_legend=False):
+    Returns: A dictionary with keys being the values (project, tissue, kit) and
+        the dictionary values being the properties (color, shape).
```

Comment:
`A dictionary with keys being the values` -> `A dictionary with keys being the DataFrame values` ?
`and the dictionary values being the properties` -> `and the dictionary values being the plotting properties` ?

Reply:
Clarified in bce269f. Let me know if it is clear.